### PR TITLE
Add a toggle for the "old" behavior of quick tags

### PR DESCRIPTION
**Pull request by ThePsionic**
This *should* add the behavior requested by @Federer343 on #150. (I'm a JavaScript newbie so if something is wrong please let me know :smile:)

### CODE REVIEW

**Comment by WyohKnott** [comment]
I don't think adding an option for that is necessary, it complicates things too much.

File: Extensions/quick_tags.js [hunk 1]
```diff
@@ -49,7 +54,8 @@ XKit.extensions.quick_tags = new Object({
 	menu_close: function() {
 		// Only close the menu if it doesn't have keyboard or mouse focus
 		if ($("#xkit-quick-tags-window").find('input:focus').length === 0 &&
-		    $('#xkit-quick-tags-window:hover').length === 0) {
+		    $('#xkit-quick-tags-window:hover').length === 0 &&
+		    Xkit.extensions.quick_tags.preferences.close_on_focus.value === false) {
```
Comment (bvtsang) [inline Extensions/quick_tags.js]: Try using `!Xkit.extensions.quick_tags.preferences.close_on_focus.value` for two reasons:

1. `Xkit` isn't defined, but `XKit` is
2. In general, if statements that are like `if (something === true)` or `if (something === false)` are redundant; instead, we can use `if (something)` or `if (!something)`

File: Extensions/quick_tags.js [hunk 2]
```diff
@@ -303,6 +309,11 @@ XKit.extensions.quick_tags = new Object({
 		$("#xkit-quick-tags-window").remove();
 		$("body").append(m_html);
 
+		if (Xkit.extensions.quick_tags.preferences.close_on_focus.value === true) {
```
Comment (bvtsang) [inline Extensions/quick_tags.js]: Like above, try using `XKit.extensions.quick_tags.preferences.close_on_focus.value`

**Comment by bvtsang** [comment]
Functionality-wise, looks good to me. However, I'm a bit unsure whether we should add this option, so I'll leave that for others to decide. (Perhaps I should've first reviewed the feature overall rather than delve into the details?)

One alternative is to have it so that we keep the popup open if it has keyboard focus, but then close it when the user presses enter.

**Comment by bvtsang** [comment]
I accepted #160 instead, because provides a simpler fix to @Federer343's request.

**Comment by Federer343** [comment]
@bvtsang That was my main issue with it really, that when I hit enter it didn't close.  So any change that fixes that will be good enough in my book.

**Pull request closed without merging**